planemo/ci.py

```diff
@@ -55,6 +55,38 @@ def filter_paths(ctx, raw_paths, path_type="repo", **kwds):
     return chunked_paths
 
 
+def changed_repos(diff_files):
+    diff_dirs = set(os.path.dirname(p) for p in diff_files)
+    diff_paths = set()
+    for diff_dir in diff_dirs:
+        new_diff_paths = set()
+        while diff_dir != "" and len(new_diff_paths) == 0:
```

Comment:
Maybe a bit more explanation on the potential problem: 

With `diff_dir != ""` I ensure that  `glob.glob(diff_dir + "/**/"` is not called for the working dir. This is needed because otherwise a change of a file in the repository root would lead to the inclusion of all tools / repos.

Hence calling from outside of the repo this would not work .. BUT this is irrelevant, because `git diff` will fail before that.

Calling from subdirs would lead to ignoring changes in this dir.


Maybe we can assert the assumption by comparing the output of `git rev-parse --show-toplevel` with the working dir.

Reply:
That said if this lets you make progress on the planemo action I'm happy to merge and iterate on it if there are issues.

Reply:
That's fine. I can easily use test cases that include changes in the tool xml. 

Just need to think of something different than `--recursive`, because this does not capture the essence. Maybe `--extended_git_diff`.